orion/pkg/controllers/location_controller.go

```diff
@@ -53,6 +55,23 @@ func CreateLocation(c *gin.Context) {
 	return
 }
 
+func PublishLocations(c *gin.Context) {
+	// Incoming JSON
+	var locIdsJson []string
+	c.BindJSON(&locIdsJson)
+
+	errors := services.LocationService.Publish(locIdsJson)
+	if len(errors) > 0 {
+		for _, err := range errors {
+			c.Error(err.Error)
+			c.String(http.StatusInternalServerError, err.Error.Error())
```

Comment:
hm it usually doesn't work like this. You can't have multiple error responses to a single request.
A better way to go about it would be to aggregate or concatenate all the errors together.
Then send the aggregate along with one internal server error.

Reply:
Just like the getAll endpoints, you can use `c.JSON(status, ...)` and create a single object which would be the JSON response body. You can pass in an array of LocationError structs - that would work.